src/xcode/ENA/ENA/Source/Scenes/Home/HomeBadgeWrapper.swift

```diff
@@ -46,6 +46,15 @@ class HomeBadgeWrapper {
 		badgesCount[badgeType] = (oldValue ?? 0) + value
 		saveAndUpdate()
 	}
+	
+	func decrease(_ badgeType: BadgeType, by value: Int) {
+		let oldValue = badgesCount[badgeType] ?? 0
+		guard oldValue ?? 0 - value > 0 else {
+			return
+		}
+		badgesCount[badgeType] = (oldValue ?? 0) - value
+		saveAndUpdate()
+	}
```

Comment:
An alternate approach to avoid those redundant `??` fallbacks could be:

```swift
func decrease(_ badgeType: BadgeType, by value: Int) {
	let oldValue = badgesCount[badgeType]?.map { $0 } ?? 0
	guard oldValue - value > 0 else { return }

	badgesCount[badgeType] = oldValue - value
	saveAndUpdate()
}
```